crates/control/src/calibration_controller.rs

```diff
@@ -0,0 +1,299 @@
+            self.current_calibration_command = new_phase;
+        }
+
+        context
+            .calibration_measurements
+            .fill_if_subscribed(|| self.current_measurements.clone());
+        context
+            .last_calibration_corrections
+            .fill_if_subscribed(|| self.last_calibration_corrections);
+        Ok(MainOutputs {
+            calibration_command: self.current_calibration_command.clone().into(),
+        })
+    }
+
+    fn get_next_command(
+        &mut self,
+        primary_state_transition_to_calibration: bool,
+        current_cycle_time: &CycleTime,
+        context: &CycleContext,
+    ) -> Option<CalibrationCommand> {
+        let look_at_dispatch_waiting = context.parameters.look_at_dispatch_wait_duration;
+        let initial_stabilization_delay = context
+            .parameters
+            .initial_to_calibration_stabilization_delay;
+        match self.current_calibration_command {
+            CalibrationCommand::INACTIVE => {
+                if primary_state_transition_to_calibration {
+                    info!(
+                        "Calibration is activated, waiting for {}s until the Robot is stable.",
+                        initial_stabilization_delay.as_secs_f32()
+                    );
+
+                    Some(CalibrationCommand::INITIALIZE {
+                        started_time: *current_cycle_time,
+                    })
+                } else {
+                    None
+                }
+            }
+            CalibrationCommand::INITIALIZE {
+                started_time: activated_time,
+            } => match context.primary_state {
+                PrimaryState::Calibration => {
+                    let waiting_duration = current_cycle_time
+                        .start_time
+                        .duration_since(activated_time.start_time)
+                        .unwrap_or_default();
+
+                    if waiting_duration >= initial_stabilization_delay {
+                        Some(self.try_transition_to_look_at(*context.cycle_time))
+                    } else {
+                        None
+                    }
+                }
+                _ => None,
+            },
+            CalibrationCommand::LOOKAT { dispatch_time, .. } => {
+                let time_diff = current_cycle_time
+                    .start_time
+                    .duration_since(dispatch_time.start_time)
+                    .unwrap_or_default();
+
+                if time_diff > look_at_dispatch_waiting {
+                    Some(CalibrationCommand::CAPTURE {
+                        dispatch_time: *current_cycle_time,
+                    })
+                } else {
+                    None
+                }
+            }
+            CalibrationCommand::CAPTURE { dispatch_time } => {
+                // TODO decide a better way to handle the two cameras. Perhaps set a capture position like in the look-at command?
```

Comment:
Are you capturing with both cameras at the same time?
If you are trying to look at a position with the top camera but VisionBottom runs once and successfully captures a measurement. Wouldn't you then transition to LookAt again and start moving the head without ever getting data from VisionTop?

Reply:
Correct, the issue I'm facing is that we might not have the right look-at list or such and not capture anything.
I'll change this to capture with specific camera (same as look-at camera). if it fails (retries exceeded), then we move on.

This means, if we don't have enough measurements, solver should abort and inform the user (i.e. audio.)

Reply:
Fixed, limited to specific camera